app/components/select_component.rb

```diff
@@ -0,0 +1,12 @@
+class SelectComponent < ApplicationComponent
+  def initialize(config, **kwargs)
+    @form = config[:form]
+    @attribute = config[:attribute]
+    @choices = config[:choices] || []
+    @options = config[:options] || {}
+    @html_options = config[:html_options] || {}
+    @skip_label = config[:skip_label] || true
```

Comment:
I think we want this to default to `false`, rather than true. Also, make sure to use [`Hash#fetch`](https://ruby-doc.org/3.2.2/Hash.html#method-i-fetch), rather  than `||`; because if `config[:skip_label]` is `false` then it will become `true`; whereas with `fetch` it only returns the default value if no value is set.

Another option would be to `def initialize(form:, attribute:, choices: [], options: {}, html_options: {}, skip_label: false)` so that each argument's defaults are part of the method definition.

Reply:
Oh good call. I actually confused myself into thinking that those arguments might come in as strings (although I even defaulted to a `true` boolean. But I had actually checked their type and forgot to come back and change these.

Reply:
Cool, does this mean you're going to lean into Ruby's assignment syntax rather than the `||` operator? As it currently stands, `SelectComponent.new(skip_label: false)` will become `@skip_label # true`.